### DISCUSSION TITLE
Support for artifacts built by webpack (static module bundler for JavaScript)

### DISCUSSION THREAD

**Discussion by sseelmann** [aquasecurity/trivy#6263] [Ideas]
### Description

Hello,

we have a JavaScript app which uses [webpack](https://webpack.js.org/concepts/) to bundle the app including its npm dependencies into a single js file. Currently Trivy isn't able to detect any vulnerable packages included, neither using `trivy fs` nor when packaged into a container image using `trivy image`.

Would it be possible to support scanning such artifacts?

Is there a workaround when doing the image scan? (adding `package-lock.json` to the image isn't considered by `trivy image`)

Thanks,
Stefan

### Target

Container Image

### Scanner

Vulnerability

**Comment by itaysk** [comment]
there are some cases where package information is lost during build time, for this reason it's recommended to scan the source code and not only the build artifacts. if you want to preserve the information you can generate SBOM from source

**Comment by nikpivkin** [comment]
@sseelmann FYI, You can use [this ](https://github.com/CycloneDX/cyclonedx-webpack-plugin)plugin to generate an SBoM that can be [scanned ](https://aquasecurity.github.io/trivy/v0.50/docs/supply-chain/sbom/#scanning)by Trivy.
